src/onnx/parse_fastgelu.cpp

```diff
@@ -0,0 +1,109 @@
+                                                                  shape::half_type};
+        auto x      = args[0];
+        auto x_type = x->get_shape().type();
+        if(not is_fast_gelu_input_type(x_type))
+        {
+            MIGRAPHX_THROW("PARSE_FASTGELU: input tensor `x` is not a float or half type");
+        }
+        if(args.size() > 1 and args.at(1)->name() != "undefined")
+        {
+            auto y      = args[1];
+            auto y_type = y->get_shape().type();
+            if(not is_fast_gelu_input_type(y_type))
+            {
+                MIGRAPHX_THROW("PARSE_FASTGELU: input tensor `bias` is not a float or half type");
+            }
+            x = info.add_common_op("add", x, y);
+        }
+
+        auto x_lens = x->get_shape().lens();
+        // FastGelu equation from
+        // https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftfastgelu
+        // Y=0.5X(1+tanh(0.797885X+0.035677XXX))
+        auto const1 = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0.797885}});
+        auto const2 = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0.035677}});
+        auto one    = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {1.0f}});
+        auto half   = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0.5f}});
+        auto three  = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {3.0f}});
+        // 0.035677XXX
+        auto three_mbcast = info.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), three);
+        auto pow0          = info.add_instruction(migraphx::make_op("pow"), {x, three_mbcast});
+        auto const2_mbcast = info.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), const2);
+        auto mul0 = info.add_instruction(migraphx::make_op("mul"), {pow0, const2_mbcast});
+
+        // 0.797885X+0.035677XXX
+        auto const1_mbcast = info.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), const1);
+        auto mul1 = info.add_instruction(migraphx::make_op("mul"), {const1_mbcast, x});
+        auto add1 = info.add_instruction(migraphx::make_op("add"), {mul0, mul1});
+
+        // 1+tanh(0.797885X+0.035677XXX)
+        auto tanh0 = info.add_instruction(migraphx::make_op("tanh"), add1);
+        auto one_mbcast =
+            info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), one);
+        auto add2 = info.add_instruction(migraphx::make_op("add"), {tanh0, one_mbcast});
+
+        // 0.5X(1+tanh(0.797885X+0.035677XXX))
+        auto half_mbcast =
+            info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), half);
+        auto mul2 = info.add_instruction(migraphx::make_op("mul"), {x, half_mbcast});
+        return info.add_instruction(migraphx::make_op("mul"), {add2, mul2});
```

Comment:
We should insert the same erf version here, our rewrite_gelu will rewrite it to the faster version. Also these two operators should use the same parser class.